modules/galera/detectors-galera.tf

```diff
@@ -0,0 +1,119 @@
+    severity              = "Critical"
+    detect_label          = "CRIT"
+    disabled              = coalesce(var.heartbeat_disabled, var.detectors_disabled)
+    notifications         = coalescelist(lookup(var.heartbeat_notifications, "critical", []), var.notifications.critical)
+    parameterized_subject = local.rule_subject
+    parameterized_body    = local.rule_body
+  }
+}
+
+resource "signalfx_detector" "wsrep_ready" {
+  name = format("%s %s", local.detector_name_prefix, "MySQL Galera wsrep_ready")
+
+  program_text = <<-EOF
+    signal = data('mysql_wsrep_ready', filter=${module.filter-tags.filter_custom})${var.wsrep_ready_aggregation_function}${var.wsrep_ready_transformation_function}.publish('signal')
+    detect(when(signal < ${var.wsrep_ready_threshold_critical})).publish('CRIT')
+EOF
+
+  rule {
+    description           = "is OFF"
+    severity              = "Critical"
+    detect_label          = "CRIT"
+    disabled              = coalesce(var.wsrep_ready_disabled, var.detectors_disabled)
+    notifications         = coalescelist(lookup(var.wsrep_ready_notifications, "critical", []), var.notifications.critical)
+    parameterized_subject = local.rule_subject
+    parameterized_body    = local.rule_body
+  }
+}
+
+resource "signalfx_detector" "wsrep_local_state" {
+  name = format("%s %s", local.detector_name_prefix, "MySQL Galera wsrep_local_state")
+
+  program_text = <<-EOF
+    signal = data('mysql_wsrep_local_state', filter=${module.filter-tags.filter_custom})${var.wsrep_local_state_aggregation_function}${var.wsrep_local_state_transformation_function}.publish('signal')
+    detect(when(signal != ${var.wsrep_local_state_threshold_critical})).publish('CRIT')
+EOF
+
+  rule {
+    description           = "is not Synced"
+    severity              = "Critical"
+    detect_label          = "CRIT"
+    disabled              = coalesce(var.wsrep_local_state_disabled, var.detectors_disabled)
+    notifications         = coalescelist(lookup(var.wsrep_local_state_notifications, "critical", []), var.notifications.critical)
+    parameterized_subject = local.rule_subject
+    parameterized_body    = local.rule_body
+  }
+}
+
+resource "signalfx_detector" "wsrep_flow_control_paused" {
+  name = format("%s %s", local.detector_name_prefix, "MySQL Galera flow_control_paused")
+
+  program_text = <<-EOF
+    signal = data('mysql_wsrep_flow_control_paused', filter=${module.filter-tags.filter_custom})${var.wsrep_flow_control_paused_aggregation_function}${var.wsrep_flow_control_paused_transformation_function}.publish('signal')
+    detect(when(signal > ${var.wsrep_flow_control_paused_threshold_critical})).publish('CRIT')
+    detect(when(signal > ${var.wsrep_flow_control_paused_threshold_major}) and when(signal < ${var.wsrep_flow_control_paused_threshold_critical})).publish('MAJOR')
+EOF
+
+  rule {
+    description           = "is > ${var.wsrep_flow_control_paused_threshold_critical}%"
```

Comment:
are you sure it is a percentage ?

Reply:
please respect usual convention (is too high ...)

Reply:
This was a WIP I changed it, here is the definition :

> When the node returns a value of 0.0, it indicates that the node did not pause due to Flow Control during this period. When the node returns a value of 1.0, it indicates that the node spent the entire period paused.